cmd/po-docgen/operator.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+)
+
+type FlagDoc struct {
+	name         string
+	defaultValue string
+	description  string
+}
+
+func fileToASTPackage(filePaths []string) *ast.Package {
+	fset := token.NewFileSet()
+	m := make(map[string]*ast.File)
+
+	for _, filePath := range filePaths {
+
+		f, err := parser.ParseFile(fset, filePath, nil, parser.ParseComments)
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+
+		m[filePath] = f
+	}
+	apkg, _ := ast.NewPackage(fset, m, nil, nil)
+
+	return apkg
+}
+
+func unquoteLiteral(value string) string {
+	if len(value) > 0 && value[0] == '"' {
+		value = value[1:]
+	}
+
+	if len(value) > 0 && value[len(value)-1] == '"' {
+		value = value[:len(value)-1]
+	}
+	return value
+}
+
+func resolveDurationExpr(expr ast.Expr) string {
+	var exprIdent *ast.Ident
+	exprSelector, ok := expr.(*ast.SelectorExpr)
+	if ok {
+		exprIdent = exprSelector.Sel
+	} else {
+		exprIdent, ok = expr.(*ast.Ident)
+		if !ok {
+			return "No Identifier to Resolve"
+		}
+
+	}
+
+	if exprIdent.Obj == nil {
+		return exprIdent.String()
+	}
+
+	retValue := ""
+
+	if exprIdent.Obj.Kind == ast.Con {
+		valSpec := exprIdent.Obj.Decl.(*ast.ValueSpec)
+		for _, value := range valSpec.Values {
+			retValue += unquoteLiteral(value.(*ast.BasicLit).Value)
+		}
+	}
+	return retValue
+
+}
+
+func resolveBoolExpr(expr ast.Expr) string {
+	exprIdent, ok := expr.(*ast.Ident)
+	if ok {
+		return exprIdent.Name
+	}
+
+	return "No Identifier to Resolve"
```

Comment:
shouldn't it return an error?

Reply:
agree, returning an error is better. anyone wants to add a new argument should also update the doc generator if needed.